troubleshooting-and-how-tos/deployment.md

```diff
@@ -0,0 +1,50 @@
+# Recipe for One-Click Deployment
+
+This document contains a brief explanation on how we have setup one-click deployment. By one-click deployment, we mean by just making a release on Github, the changes are deployed without any human action. Many of the items used from the AWS menu are explained below.
+
+- **<u>Docker</u>**: 
+
+  Both the frontend and backend are containerized in a node-14 base image, with a Dockerfile akin to this [one](https://github.com/rakaar/dockerize-react-and-express-app/blob/master/Dockerfile).
+
+- **<u>Github Actions</u>**: 
+
+  On release, a Github Action is triggered. The YAML file action has been inspired from the [C-3PO project from LTTKGP](https://github.com/lttkgp/C-3PO/blob/master/.github/workflows/deploy.yml). This Actions in brief do the following (The AWS terms will be explained soon)
+
+  - Logging into AWS using the secrets configured via Github Secrets
+  - Builds and pushes the image to **Elastic Container Registry(ECR)**
+  - Deploy the image via **Elastic Container Service(ECS) Fargate**
+
+- <u>**Elastic Container Registry(ECR)**</u>: 
+
+  AWS Service where you can store Docker Container Images. The Docker images are stored in a private repository. To look at the images in AWS, visit ECR by searching in the console. Click on the repository name. 
+
+- **<u>Elastic Container Service(ECS) Fargate</u>**: 
+
+  Once the latest image is stored in the ECR, they need to be deployed. A service by name **ECS Fargate** handles this job of deploying the docker image. ECS Fargate is configured using a JSON file present in the `deploy` folder. The JSON is obtained from the AWS Console(Instructions to obtain JSON from AWS console are pinned in Felvin Slack). 
+
+  The following are 4 important things related to ECS Fargate. The below diagram will help you get an idea of different layers of Fargate. To setup a ECS Fargate service, we need to configure the below four things.
+
+  ![](https://imgur.com/clXWtjU.png)
+
+1. **Container Defintion**: Details of the Docker container. We need to specify the docker Image URI, which is already existing in ECR.
+
+2. **Task definition:** Details of the machine where Image is deployed. Things like Memory size, CPU units, Network mode are specified. The Network mode is AWS **Virtual Private Cloud(VPC)**. In VPC, we have 2 subnets, where our images are deployed.
+
+3. **Service**: A service is a collection of Tasks. The service can be of Fargate type(like ours) or EC2 type  or External type. We are using Fargate Service because on our behalf it defines resources, manages and runs the resources. The below image explains better.
```

Comment:
I'm not sure if this should be called a collection of tasks. A service takes a task definition and adds more details (Such as the VPC, subnets, ports, load balancers) and schedules it to be run. AFAIK one service runs just one task definition. The service controls how many running instances do you want (Minimum and maximum instances that should exist at a given time), health checks to be configured, the load balancer to use, and the production port to bind to. 

I think what you want to mention is that a service controls how many tasks will exist. A task is an instance of a service, which is a recipe. A task is an actual container running the task definition we've provided, and exposing the ports mentioned

It would be better to separate task definition from task, to avoid this confusion IMO

Reply:
Thanks, added the suggestions